server/conn.go

```diff
@@ -1456,8 +1459,23 @@ func (cc *clientConn) handleStmt(ctx context.Context, stmt ast.StmtNode, warns [
 	reg := trace.StartRegion(ctx, "ExecuteStmt")
 	rs, err := cc.ctx.ExecuteStmt(ctx, stmt)
 	reg.End()
+	// The session tracker detachment from global tracker is solved in the `rs.Close` in most cases.
+	// If the stmt have no rs like `insert`, The detachment work will be directly
+	// done in the `defer` function.
 	if rs != nil {
 		defer terror.Call(rs.Close)
+	} else {
+		sc := cc.ctx.GetSessionVars().StmtCtx
+		defer func() {
+			if sc != nil {
+				if sc.MemTracker != nil {
+					sc.MemTracker.DetachFromGlobalTracker()
+				}
+				if sc.DiskTracker != nil {
+					sc.DiskTracker.DetachFromGlobalTracker()
+				}
+			}
+		}()
```

Comment:
Emmmm, can we put this logic in `seesion.ExecuteStmt()`?
Since code in `conn.go` is about dealing with the network, it's better to not put it here.

Reply:
What're your options? @SunRunAway @Yisaer

Reply:
Why not put them into [handleNoDelay](https://github.com/pingcap/tidb/blob/915d84d7ad4a0335df8479dacce555b68d7363ac/executor/adapter.go#L372)?

Reply:
updated.